### ISSUE TITLE
Use first argument of callback instead of throw on error

### ISSUE THREAD

**Issue by Rundfunk** [Leonidas-from-XIV/node-xml2js#67]
Hi Leonidas,

While working with xml2js, I found that the first argument of the callback is not used when an error is encountered (like parsing incorrect XML), instead `throw` is used. (See line 183 of `xml2js.coffee`.) 

Why is an error thrown? This requires the usage of a `try/catch` block, instead of checking if `err` is set.

Let's say I have the following code where I check the first argument of the callback:

``` javascript
parseString xml, (err, result) ->
    console.log 'encountered error' if err
```

This does not work at the moment, causing the need for a try/catch block:

``` javascript
try {
    parseString xml, (err, result) ->
        # process result
} catch (e) {
    console.log 'encountered error', e
}
```

However, when the `# process result` becomes very long, the code for which the `try` is needed and the `catch` become distanced from each other, making it difficult to read.

To summarize my question: _can the first argument of the callback be used to communicate an error instead of throwing one?_

**Comment by jcsahnwaldt** [comment]
Looks like this issue still occurs in xml2js 0.4.19.

```js
const xml2js = require('xml2js');
let xml = `<a><b>x</a></b>`;
try {
  xml2js.parseString(xml, (err, res) => {
    console.log('callback err:');
    console.log(err);
    console.log('callback res:');
    console.log(res);
  });
}
catch (err) {
  console.log('caught err:');
  console.log(err);
}
```
Output (paths in stacktrace sanitized):
```
callback err:
Error: Unexpected close tag
Line: 0
Column: 11
Char: >
    at error (./node_modules/sax/lib/sax.js:651:10)
    at strictFail (./node_modules/sax/lib/sax.js:677:7)
    at closeTag (./node_modules/sax/lib/sax.js:871:9)
    at SAXParser.write (./node_modules/sax/lib/sax.js:1436:13)
    at Parser.exports.Parser.Parser.parseString (./node_modules/xml2js/lib/parser.js:322:31)
    at Parser.parseString (./node_modules/xml2js/lib/parser.js:5:59)
    at Object.exports.parseString (./node_modules/xml2js/lib/parser.js:354:19)
    at Object.<anonymous> (./test.js:6:10)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
callback res:
undefined
caught err:
Error: Unmatched closing tag: b
Line: 0
Column: 15
Char: >
    at error (./node_modules/sax/lib/sax.js:651:10)
    at strictFail (./node_modules/sax/lib/sax.js:677:7)
    at closeTag (./node_modules/sax/lib/sax.js:879:7)
    at SAXParser.write (./node_modules/sax/lib/sax.js:1436:13)
    at Parser.exports.Parser.Parser.parseString (./node_modules/xml2js/lib/parser.js:322:31)
    at Parser.parseString (./node_modules/xml2js/lib/parser.js:5:59)
    at Object.exports.parseString (./node_modules/xml2js/lib/parser.js:354:19)
    at Object.<anonymous> (./test.js:6:10)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
```
First, the callback function is called with an `err` parameter, but then an exception is thrown with a slightly different `err` object.
